sys-apps/edac-utils/edac-utils-0.18-r1.ebuild

```diff
@@ -0,0 +1,75 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit autotools flag-o-matic toolchain-funcs
+
+DESCRIPTION="Userspace helper for Linux kernel EDAC drivers"
+HOMEPAGE="https://github.com/grondo/edac-utils"
+SRC_URI="https://github.com/grondo/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="debug"
+
+DEPEND="sys-fs/sysfsutils"
+RDEPEND="${DEPEND}
+	sys-apps/dmidecode"
+
+src_prepare() {
+	default
+
+	# <sys-devel/gcc-10, <sys-devel/binutils-2.34 do not support LDPT_ADD_SYMBOLS_V2
+	# https://sourceware.org/bugzilla/show_bug.cgi?id=25355
+	# We use Portage's build environment to determine which GCC and binutils version are
+	# active. If either fail to meet the minimum version, -flto* is stripped from FLAGS
+	local gcc_vers=$(gcc-version)
+	local ld_vers=$($(tc-getLD) -v | sed 's/.*\(2\.[0-9][0-9]\).*/\1/')
```

Comment:
This regex fails with binutils `2.9` (or any version where minor version consists of single digit) and `3.0` (or any version where major version is not `2`).

Reply:
Thanks for the feedback! I understand these are more hacks than fixes, but I couldn't figure a way that would be compatible with both bfd and gold. what I had in mind is that Gentoo wouldn't use a binutils with a single digit minor version and it would be some years before 3.0. What would be the proper way to do a version check?

Reply:
In regular expressions, `[0-9][0-9]` are exactly 2 digits, while `[0-9]+` would be any non-zero number of digits, so it would handle `2.5`, `2.100` etc.
Specifically for `sed` tool, `+` needs to be escaped, so `[0-9]\+`.
For correct handling of major version, expectation of space before major version should be added.
Updated full line:
```
local ld_vers=$($(tc-getLD) -v | sed 's/.* \([0-9]\+\.[0-9]\+\).*/\1/'
```